### PR TITLE
Make it possible to create new text themes

### LINKED ISSUE #5916: Possible to customize TextTheme?
In our leafy app, we want to build a customized TextTheme that has some different font size and font weight than material design. Is there anyway we can achieve this?

### PR DESCRIPTION
**Pull request by abarth**
Fixes #5916

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -116,8 +116,10 @@ class TextStyle {
   /// If the underlying values are null, then the corresponding factors and/or
   /// deltas must not be specified.
   ///
-  /// The non-numeric fields are copied verbatim.
+  /// The non-numeric fields can be controlled using the cooresponding arguments.
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_style.dart]: too many os

**Comment by Hixie** [comment]
LGTM but I feel that the documentation should frown on this more, given that it basically violates the material spec to do this.

**Comment by abarth** [comment]
Ok.  I'll add some more text that points to the spec.

**Pull request merged**